### ISSUE TITLE
e.getSession is not a function after authenticating using withAuthenticator with federated identity

### ISSUE THREAD

**Issue by pnewhook** [aws-amplify/amplify-js#7554]
**Describe the bug**
After setting up a new application and authenticating using `withAuthenticator(App, { federated: { facebookAppId: "XYZ" },});` the following error is printed to the browser console after login. This is happening in both my real application and a test app I just created.

```
Auth.ts:1389 Uncaught (in promise) TypeError: e.getSession is not a function
    at Auth.ts:1389
    at new Promise (<anonymous>)
    at e.userSession (Auth.ts:1387)
    at Auth.ts:1119
    at new Promise (<anonymous>)
    at e.userAttributes (Auth.ts:1118)
    at e.verifiedContact (Auth.ts:1133)
    at auth-helpers-442d4bea.js:56
    at auth-helpers-442d4bea.js:32
    at Object.next (auth-helpers-442d4bea.js:13)
```

**To Reproduce**
1. Create new react typescript app using cra
2. `amplify init`
3. Setup hosting and authentication using a social provider (facebook in my case)
4. Wrap app in HOC `withAuthenticator`
5. `amplify publish`
6. View app and signin with valid user account
7. Look at chrome console

**Expected behavior**
User should be logged in without errors

**What is Configured?**
* Followed steps on https://docs.amplify.aws/lib/auth/social/q/platform/js
* Configured hosting and auth

```
/* eslint-disable */
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
    "aws_project_region": "us-east-1",
    "aws_cognito_identity_pool_id": "us-east-1:*****",
    "aws_cognito_region": "us-east-1",
    "aws_user_pools_id": "us-east-1_*****",
    "aws_user_pools_web_client_id": "*****",
    "oauth": {
        "domain": "amplifytest*****-dev.auth.us-east-1.amazoncognito.com",
        "scope": [
            "phone",
            "email",
            "openid",
            "profile",
            "aws.cognito.signin.user.admin"
        ],
        "redirectSignIn": "https://dev.*****.amplifyapp.com/",
        "redirectSignOut": "https://dev.*****.amplifyapp.com/",
        "responseType": "code"
    },
    "federationTarget": "COGNITO_USER_POOLS"
};


export default awsmobile;

   ```

<details>
  <summary><strong>Environment</strong></summary>

```
npx: installed 1 in 1.485s

  System:
    OS: Windows 10 10.0.19042
    CPU: (4) x64 Intel(R) Core(TM) i5-4670 CPU @ 3.40GHz
    Memory: 8.50 GB / 15.89 GB
  Binaries:
    Node: 14.15.3 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.9 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.423.0), Chromium (87.0.664.75)
    Internet Explorer: 11.0.19041.1
  npmPackages:
    @aws-amplify/ui-react: ^0.2.34 => 0.2.34
    @testing-library/jest-dom: ^5.11.9 => 5.11.9
    @testing-library/react: ^11.2.3 => 11.2.3
    @testing-library/user-event: ^12.6.0 => 12.6.0
    @types/jest: ^26.0.20 => 26.0.20
    @types/node: ^12.19.13 => 12.19.13
    @types/react: ^16.14.2 => 16.14.2
    @types/react-dom: ^16.9.10 => 16.9.10
    aws-amplify: ^3.3.14 => 3.3.14
    react: ^17.0.1 => 17.0.1
    react-dom: ^17.0.1 => 17.0.1
    react-scripts: 4.0.1 => 4.0.1
    typescript: ^4.1.3 => 4.1.3
    web-vitals: ^0.2.4 => 0.2.4
  npmGlobalPackages:
    @aws-amplify/cli: 4.41.1
    npm: 6.14.9
    ts-node: 9.1.1
    typescript: 4.1.3
```

</details>

**Additional context**

**_You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app._**

Edit by @iartemiev: redacted customer info in aws-exports and deleted previous revision from GitHub history

**Comment by amhinson** [comment]
Are you using the `@aws-amplify/ui-react` library? If so, the `withAuthenticator` HOC [does not support](https://docs.amplify.aws/ui/auth/authenticator/q/framework/react#breaking-changes-for-withauthenticator) additional properties passed in the 2nd argument. The alternative is to use the `AmplifyAuthenticator` [component](https://docs.amplify.aws/ui/auth/authenticator/q/framework/react#basic-usage) to wrap your application, and you can also pass the `federated` [prop](https://docs.amplify.aws/ui/auth/authenticator/q/framework/react#props-attr-amplify-authenticator) there.

**Comment by pnewhook** [comment]
I'm still getting the same error after switching to AmplifyAuthenticator. AmplifyAuthenticator also appears to behave differently. Using withAuthenticator my application wasn't loaded until the user signed in. But after changing I can see API requests on the landing page are being fired and failing because they're unauthenticated. If you direct message me I can provide our test site for you to see.

Can the typings be updated? TypeScript shouldn't have allowed me to pass that argument.
The docs probably need an update as well, https://docs.amplify.aws/ui/auth/authenticator/q/framework/react#withauthenticator.

**Comment by amhinson** [comment]
With the `AmplifyAuthenticator`, you'll need to setup your app to utilize the `AuthState` like in the [recommended usage](https://docs.amplify.aws/ui/auth/authenticator/q/framework/react#recommended-usage), but I think I misspoke regarding the breaking changes for `withAuthenticator`, as `federated` is actually a prop of `AmplifyAuthenticator` which means it can be passed in the 2nd argument (as you are doing).

With that said, I'll work on trying to reproduce this 👍

**Comment by amhinson** [comment]
Thanks again for bringing this to our attention! It has been addressed in #7562 and is available for testing with `@aws-amplify/ui-react@unstable`. You will also need to use `aws-amplify@unstable` to test this. Let me know if you're still having issues! We should be getting a new release out later this week.

**Comment by vitaliikravets** [comment]
I upgraded @aws-amplify/ui-react to version 1.1.0 and still get this error. Is there a version where it is fixed?
